modin/experimental/xgboost/xgboost_ray.py

```diff
@@ -339,67 +432,58 @@ def _train(
 
         # Train
         fut = [
-            actor.train.remote(rabit_args, params, *args, **kwargs)
-            for _, actor in actors.items()
+            actor.train.remote(rabit_args, params, *args, **kwargs) for actor in actors
         ]
-
         # All results should be the same because of Rabit tracking. So we just
         # return the first one.
         result = ray.get(fut[0])
+        remove_placement_group(pg)
         LOGGER.info(f"Training time: {time.time() - s} s")
         return result
 
 
 def _predict(
     booster,
     data,
-    nthread: Optional[int] = cpu_count(),
+    num_actors,
     **kwargs,
 ):
     s = time.time()
 
-    X, _ = data
-    X_row_parts = unwrap_partitions(X, axis=0, get_ip=True)
+    X_row_parts, _ = data
 
-    # Create remote actors
-    actors = create_actors(nthread=nthread)
+    num_actors = _get_num_actors(
+        num_actors if isinstance(num_actors, int) else "default_predict"
+    )
+
+    if num_actors > len(X_row_parts):
+        num_actors = len(X_row_parts)
 
-    assert len(actors) <= len(
-        X_row_parts
-    ), f"{len(X_row_parts)} row partitions couldn't be distributed between {len(actors)} nodes."
+    # Create remote actors
+    actors, pg = create_actors(num_actors)
 
     # Split data across workers
-    order_of_parts = _split_data_across_actors(
+    _split_data_across_actors(
         actors,
         lambda actor, *X: actor.set_predict_data.remote(*X),
         X_row_parts,
+        is_predict=True,
     )
 
     LOGGER.info(f"Data preparation time: {time.time() - s} s")
     s = time.time()
 
-    # Predict
+    booster = ray.put(booster)
+
     predictions = [
-        actor.predict._remote(
-            args=(booster,), kwargs=kwargs, num_returns=len(order_of_parts[ip])
-        )
-        if len(order_of_parts[ip]) > 1
-        else [
-            actor.predict._remote(
-                args=(booster,), kwargs=kwargs, num_returns=len(order_of_parts[ip])
-            )
-        ]
-        for ip, actor in actors.items()
+        tuple(actor.predict._remote(args=(booster,), kwargs=kwargs, num_returns=2))
+        for actor in actors
     ]
 
-    results_to_sort = list()
-    for ip, part_res in zip(actors, predictions):
-        results_to_sort.extend(list(zip(part_res, order_of_parts[ip])))
-
-    results = sorted(results_to_sort, key=lambda l: l[1])
-    results = [part_res for part_res, _ in results]
+    ray.wait([part for _, part in predictions], num_returns=len(predictions))
```

Comment:
Shouldn't we assign returned objects to `partitions` and then create DataFrame from them?